### ISSUE TITLE
[MapRequestPayload] Fail because property documentation

### ISSUE THREAD

**Issue by GErpeldinger** [symfony/symfony#58719]
### Symfony version(s) affected

6.4

### Description

I was using `#[MapRequestPayload]` to fill a DTO. In this DTO, I have an array property that is populated using explode in the constructor.

Everything worked fine until PHPStan prompted me to describe the array property. After adding the annotation `/** @var string[] */`, a ValidationFailedException occurred, indicating "This value should be of type array."

Interestingly, replacing `@var` with `@phpstan-var` resolves the issue, suggesting that Symfony enforces the type based on the `@var` annotation ? But it seems to not working correctly in some case.

### How to reproduce

```php
<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\HttpKernel\Attribute\MapRequestPayload;
use Symfony\Component\Routing\Attribute\Route;

#[Route('bug')]
class BugController extends AbstractController
{
    public function __invoke(
        #[MapRequestPayload] BugDTO $dto
    ): JsonResponse {
        return $this->json($dto);
    }
}

class BugDTO
{
    /** @var string[] The culprit is here */
    public readonly array $items;

    public function __construct(
        string $items
    ) {
        $this->items = explode(',', $items);
    }
}
```

**Comment by vltrof** [comment]
Greetings! 
It seems like when normalizer parse DTO name overlapping  occurs and produce this undefined behavior
You have constructor param and property with same name but different types and normalizer look at property type definition then constructor param
If change the property name everything works fine
for example
```php
<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\HttpKernel\Attribute\MapRequestPayload;
use Symfony\Component\Routing\Attribute\Route;

#[Route('bug')]
class BugController extends AbstractController
{
    public function __invoke(
        #[MapRequestPayload] BugDTO $dto
    ): JsonResponse {
        return $this->json($dto);
    }
}

class BugDTO
{
    /** @var string[] The culprit is here */
    public readonly array $list;

    public function __construct(
        string $items
    ) {
        $this->list = explode(',', $items);
    }
}
```

So the simplest fix is change property name, at least to make your code works

**Comment by GErpeldinger** [comment]
Hello, thank you for your response! You’re right; that’s also a solution I tested. However, as I mentioned in my first message, I prefer using `@phpstan-var` instead of `@var`. This approach allows me to keep the same variable/property names without issues.

Since this behavior didn’t seem normal to me, I decided to open an issue about it. If it isn’t resolved, this issue can at least serve as a reference for other developers facing the same problem.

Best regards.
